[webkitpy] Add "keep alive" option to SimulatedDeviceManager.initialize_devices.

bls1999 opened the pull request:
#### a77b026ded7b2928ff8974317c2bc91cb91c8d97
<pre>
[webkitpy] Add &quot;keep alive&quot; option to SimulatedDeviceManager.initialize_devices.
<a href="https://bugs.webkit.org/show_bug.cgi?id=273259">https://bugs.webkit.org/show_bug.cgi?id=273259</a>
<a href="https://rdar.apple.com/127061752">rdar://127061752</a>

Reviewed by Jonathan Bedard.

Simulated devices are currently destroyed after webkitpy exits. This has served our
use cases until this point; however, in the interest of making webkitpy able to
handle more generalized use cases, this PR adds a &quot;keep alive&quot; option to
SimulatedDeviceManager.initialize_devices. This will allow scripts to invoke this
method with the intention of keeping the initialized devices running after script
exit.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDeviceManager.initialize_devices): Add keep_alive parameter which skips registration of SimulatedDeviceManager.tear_down as the exit method.

Canonical link: <a href="https://commits.webkit.org/278010@main">https://commits.webkit.org/278010@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7acff6e4b3dc67a48e0305de3200d21dc0afee5e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49175 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28454 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52201 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51972 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45274 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/51478 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34451 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/26050 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40189 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/51041 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42382 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21302 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/49034 "Passed tests") | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/43559 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7498 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/44064 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53883 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24257 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20455 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/47509 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25530 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42587 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/46495 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10825 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26359 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25251 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/webkitpy/xcode/simulated_device.py:
```diff
@@ -403,7 +403,10 @@ def initialize_devices(cls, requests, host=None, name_base='Managed', simulator_
         SimulatedDeviceManager.INITIALIZED_DEVICES = []
-        atexit.register(SimulatedDeviceManager.tear_down)
+
+        # Keep simulators alive after exit?
```
Let's just get rid of this comment entirely.

bls1999 replied:
Done!

bls1999 committed:
[webkitpy] Add "keep alive" option to SimulatedDeviceManager.initialize_devices.
https://bugs.webkit.org/show_bug.cgi?id=273259
rdar://127061752

Reviewed by Jonathan Bedard.

Simulated devices are currently destroyed after webkitpy exits. This has served our
use cases until this point; however, in the interest of making webkitpy able to
handle more generalized use cases, this PR adds a "keep alive" option to
SimulatedDeviceManager.initialize_devices. This will allow scripts to invoke this
method with the intention of keeping the initialized devices running after script
exit.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDeviceManager.initialize_devices): Add keep_alive parameter which skips registration of SimulatedDeviceManager.tear_down as the exit method.

Canonical link: https://commits.webkit.org/278010@main
